### PR TITLE
Update the template of using Spinnaker to deploy the Azure VMSS

### PR DESCRIPTION
**Pull request by mybayern1974**
The 301-jenkins-aptly-spinnaker-vmss template has been broken for a while. This PR is to fix it.

### CODE REVIEW

File: 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json [hunk 1]
```diff
@@ -55,8 +61,7 @@
     "publicIPAddressName": "[concat(variables('scenarioPrefix'), 'PublicIP')]",
     "vmName": "[concat(variables('scenarioPrefix'), 'VM')]",
     "vmExtensionName": "[concat(variables('scenarioPrefix'), 'Init')]",
-    "spinnakerOSDiskName": "[concat(variables('scenarioPrefix'), 'OSDisk')]",
-    "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-devops-utils/v0.24.0/",
+    "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-devops-utils/master/",
```
Comment (bmoore-msft) [inline 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts - probably need to refactor this a bit to comply...

File: 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json [hunk 2]
```diff
@@ -134,6 +139,17 @@
...
           ]
+        },
+        {
+          "type": "secrets",
+          "name": "VMSshPublicKey",
+          "apiVersion": "2015-06-01",
+          "properties": {
+            "value": "[parameters('sshPublicKey')]"
+          },
+          "dependsOn": [
```
Comment (bmoore-msft) [inline 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json]: move this above properties

File: 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json [hunk 3]
```diff
@@ -329,7 +345,7 @@
...
+              "commandToExecute": "[concat('./', variables('_extensionScript'), ' -ju \"', parameters('adminUsername'), '\" -jp ''', parameters('adminPassword'), ''' -ai \"', parameters('servicePrincipalAppId'), '\" -ak \"', parameters('servicePrincipalAppKey'), '\" -ti \"', subscription().tenantId, '\" -si \"', subscription().subscriptionId, '\" -rg \"', resourceGroup().name, '\" -vn \"', variables('keyVaultName'), '\" -san \"', variables('storageAccountName'), '\" -sak \"', listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2016-01-01').keys[0].value, '\" -vf \"', reference(variables('publicIPAddressName')).dnsSettings.fqdn, '\" -r \"', parameters('location'), '\" -al \"', variables('_artifactsLocation'), '\" -st \"', variables('_artifactsLocationSasToken'), '\"')]"
```
Comment (bmoore-msft) [inline 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json]: update apiVersion

File: 301-jenkins-aptly-spinnaker-vmss/README.md [hunk 4]
```diff
@@ -7,13 +7,14 @@
   <img src="http://armviz.io/visualizebutton.png"/>
 </a>
 
-This template allows you to deploy and configure a DevOps pipeline from an Aptly repository to a VM Scale Set in Azure. It deploys an instance of Jenkins and Spinnaker on a Linux Ubuntu 14.04 LTS VM.
+This template allows you to deploy and configure a DevOps pipeline from an Aptly repository to a VM Scale Set in Azure. It deploys an instance of Jenkins and Spinnaker on a Linux Ubuntu 16.04 LTS VM.
 
-The Jenkins instance will include a basic pipeline that checks out a [sample git repository](https://github.com/azure-devops/hello-karyon-rxnetty.git), builds the debian package, and pushes the package to an Aptly repository hosted on the VM. The Spinnaker instance will automatically be setup to listen to that Jenkins instance and to deploy VM Scale Sets.
+The Jenkins instance will include a basic pipeline that checks out a [sample git repository](https://github.com/mybayern1974/hello-karyon-rxnetty.git), builds the debian package, and pushes the package to an Aptly repository hosted on the VM. The Spinnaker instance will automatically be setup to listen to that Jenkins instance and to deploy VM Scale Sets.
```
Comment (bmoore-msft) [inline 301-jenkins-aptly-spinnaker-vmss/README.md]: need to point this to a "org" repo, not a personal one...

File: 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json (hunk 1 above)
Comment (mybayern1974) [reply 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json]: @bmoore-msft , thank you for your review! Updated content for most of your comments. While as for this one that suggests to follow the BP guide, I've initiated an offline discussion with you. Based on that would you please share some best practices?

**Pull request merged**